src/clarity-angular/data/tree-view/abstract-tree-selection.ts

```diff
@@ -76,18 +81,28 @@ export abstract class AbstractTreeSelection {
         if (this.selected !== previousSelectedValue) {
             this.selectedChanged();
         }
+
+        if (this.indeterminate !== previousIndeterminateValue) {
+            this.indeterminateChanged();
+        }
     }
 
     parentChanged(selected: boolean) {
         if (selected && !this.selected) {
             this._selected = true;
-            this._indeterminate = false;
+            if (this._indeterminate) {
+                this._indeterminate = false;
+                this.indeterminateChanged();
+            }
```

Comment:
Why are you manually re-writing the same exact code as `set indeterminate()` everywhere? Might as well use it, right? I see at least 3 places where you're doing this.